### ISSUE TITLE
stex-importer does not deal with OMDoc files that are not theories

### ISSUE THREAD

**Issue by kohlhase** [UniFormal/MMT#223]
I am looking at the OMDoc/MMT generated by my new AI course materials. And in many slides, there is not a top-level theory, e.g. in `AI/source/propositional-reasoning/en/BNI-enum-principle.tex` we get an OMDoc file from LaTeXML, but then the `stex-importer` only generates `AI/probabilistic-reasoning/en/BNI-enum-principle.omdoc/omdoc1.omgroup1/_.omdoc`, which only contains
```XML
<omdoc xmlns="http://omdoc.org/ns" xmlns:om="http://www.openmath.org/OpenMath">
<theory name="_" base="http://mathhub.info/MiKoMH/AI/probabilistic-reasoning/en/BNI-enum-principle.omdoc/omdoc1.omgroup1">
</theory>
</omdoc>
```
This is clearly wrong. 
As I have a student who wants to work on guided tours, I would like to have some materials (best the current courses) to work on.

**Comment by kohlhase** [comment]
I see that there is another problem, there are nested `<omtext>` elements in the LaTeXML-generate OMDoc, but I think that is not the problem. And I can fix that.

**Comment by m-iancu** [comment]
@kohlhase Is the OMDoc generated by LaTeXML in this case anywhere ? I could take a quick look and see if there are unexpected top-level elements that could throw off the importer.

**Comment by kohlhase** [comment]
[BNI-enum-principle.txt](https://github.com/UniFormal/MMT/files/1134585/BNI-enum-principle.txt)

here it is, as I said above, there is a nested <omtext> which is still wrong, I suspect that the <omdoc:uses> might throw it off as well (do you have some kind of treatment for that?

**Comment by m-iancu** [comment]
Is the omdoc:uses tag new ? The importer knows omdoc:imports instead. I
think this causes a lot of other parsing to fail because imported symbols
are not found. I think this should be a quick change on the importer side
and might fix the big issue (i.e. effectively empty files).

**Comment by kohlhase** [comment]
omdoc:uses is relatively new, but I have been using it for a two years at least. 

It is a kind of narrative theory import that is not re-exported. It can appear in any narrative context and makes the symbols from the referenced theory available.

**Comment by m-iancu** [comment]
Yes, i remember now but I don't think it was there in SMGloM (but maybe I
misremember). Either way, it's not supported in the importer. A quick fix
would be to change
https://github.com/UniFormal/MMT/blob/master/src/mmt-stex/src/info/kwarc/mmt/stex/STeXImporter.scala#L257
to
"case "imports" | "uses" => ". Later one could add more specific handling
for it, but for making symbols available during parsing an include will do.

**Comment by kohlhase** [comment]
yes, that seems sensible. Would you do that in the devel branch?

**Comment by m-iancu** [comment]
Done. I didn't test, but if there are issues I can follow up.

**Comment by kohlhase** [comment]
thanks, I will test this when I have prepared all my CICM talks.

**Comment by tkw1536** [comment]
This issue is still assigned to me, but seems very old. Probably fixed already with all the new versions of STeX since, so I'm closing it.
